main.js

```diff
@@ -1,3 +1,110 @@
 import './style.css';
 
-console.log('Hello world');
+/**
+ * Creates tooltip label (triangle and box)
+ * @param {string} tooltipText
+ * @returns {HTMLElement}
+ */
+const makeTooltipElems = (tooltipText) => {
+    const triangle = document.createElement('div');
+    triangle.classList.add('tooltip__triangle');
+    const box = document.createElement('div');
+    box.classList.add('tooltip__box');
+    box.textContent = tooltipText || 'No tooltip!';
+    box.append(triangle);
+
+    return box;
+};
+
+document.querySelectorAll('.tooltip').forEach((item) => {
+    const tooltipBox = makeTooltipElems(item.dataset.tooltip);
+    item.append(tooltipBox);
+});
+
+/**
+ * Toggles visibility and opacity on element
+ * @param {elem: HTMLElement} elem 
+ */
+const toggleVisibility = (elem) => {
+  if (elem.isHidden) {
+    elem.style.visibility = "visible";
```

Comment:
it is better to use `display: none` to hide elements. If you need to animate it, you can describe an animation to the class where you will animate opacity and then hide by `display: none`